tests/binaries/heap_find_fake_fast.c

```diff
@@ -10,17 +10,91 @@
  */
 
 #include <stdlib.h>
+#include <assert.h>
 
 void break_here(void) {}
 
 // Fake chunk size field with a set NON_MAIN_ARENA flag.
 // Enough space afterwards to ensure only this fake size field is a candidate.
 char fake_chunk[0x80] __attribute__((aligned(0x10))) = "XXXXXXXX\x7f";
 
-int main(void)
-{
+// This buffer will contain the fake chunk sizes
+unsigned long buf[64] __attribute__((aligned(0x10)));
+
+// This is the address we want the fake chunks to overlap with
+unsigned long target_address;
+
+/**
+ * Put the value of `size` at `distance` bytes before the address of
+ * `target_address`
+ */
+void setup_mem(unsigned long size, unsigned distance) {
+    memset(buf, 0, sizeof(buf));
+    target_address = 0;
+
+    char *chunk_size_addr = (char*)&target_address - distance;
+    *(unsigned long*)chunk_size_addr = size;
+}
+
+int main(void) {
+    assert((unsigned long)&target_address - (unsigned long)buf == sizeof(buf));
     // Initialize malloc so heap commands can run.
     void* m = malloc(0x18);
 
+    // A valid aligned fastbin chunk with no flags set
+    setup_mem(0x20, 0x8);
+    break_here();
+
+    // A valid aligned fastbin chunk with all flags set
+    setup_mem(0x2F, 0x8);
+    break_here();
+
+    // A valid unaligned fastbin chunk
+    setup_mem(0x20, 0x9);
+    break_here();
+
+    // A valid aligned fastbin chunk that's too close to the target address (the
+    // size overlaps the target address)
+    setup_mem(0x20, 0x0);
+    break_here();
+
+    // A valid unaligned fastbin chunk that's too close to the target address (the
+    // size overlaps the target address)
+    setup_mem(0x20, 0x7);
+    break_here();
+
+    // An invalid chunk with a size below the minimum chunk size
+    setup_mem(0x1F, 0x8);
+    break_here();
+
+    // A valid aligned fastbin chunk just in range of the target address
+    setup_mem(0x80, 0x78);
+    break_here();
+
+    // A valid unaligned fastbin chunk just in range of the target address
+    /* setup_mem(0x80, 0x7F); */
```

Comment:
I would have thought this would result in a 1 byte partial overwrite of the target, but the command isn't finding it.